src/query/storages/fuse/src/io/read/block_reader.rs

```diff
@@ -344,6 +344,39 @@ impl BlockReader {
         futures::future::try_join_all(join_handlers).await
     }
 
+    pub fn support_blocking_api(&self) -> bool {
+        self.operator.metadata().can_blocking()
+    }
+
+    pub fn sync_read_columns_data(&self, part: PartInfoPtr) -> Result<Vec<(usize, Vec<u8>)>> {
+        let part = FusePartInfo::from_part(&part)?;
+
+        let columns = self.column_leaves.get_by_projection(&self.projection)?;
+        let indices = Self::build_projection_indices(&columns);
+
+        let mut threads = Vec::with_capacity(indices.len());
+        let mut results = Vec::with_capacity(indices.len());
+
+        for index in indices {
+            let column_meta = &part.columns_meta[&index];
+
+            let op = self.operator.clone();
+
+            let location = part.location.clone();
+            let offset = column_meta.offset;
+            let length = column_meta.length;
+
+            threads.push(std::thread::spawn(move || {
```

Comment:
std::thread -> common_base::base::Thread

Reply:
Maybe we can use https://github.com/rayon-rs/rayon?

Reply:
common_base::base::Thread is used for resource tracking.